File: GridView/EFCodeFirstInHierarchy/RadGridViewEFCodeFirstInHierarchy/RadGridViewEFCodeFirst.Models/OrderType.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RadGridViewEFCodeFirst.Models
{
    public class OrderType
    {
        private ICollection<Order> orders;
        private ICollection<Shipper> shippers;

        public OrderType()
        {
            this.orders = new HashSet<Order>();
            this.shippers = new HashSet<Shipper>();
        }

        public int OrderTypeId { get; set; }

        public virtual ICollection<Order> Orders
        {
            get
            {
                return this.orders;
            }
            set
            {
                this.orders = value;
            }
        }

        public virtual ICollection<Shipper> Shippers
        {
            get
            {
                return this.shippers;
            }
            set
            {
                this.shippers = value;
            }
        }

        public string  Description { get; set; }
    }
}

File: GridView/EFCodeFirstInHierarchy/RadGridViewEFCodeFirstInHierarchyVB/RadGridViewEFCodeFirst.Models/OrderType.vb
Public Class OrderType
    Private m_orders As ICollection(Of Order)
    Private m_shippers As ICollection(Of Shipper)

    Public Sub New()
        Me.m_orders = New HashSet(Of Order)()
        Me.m_shippers = New HashSet(Of Shipper)()
    End Sub

    Public Property OrderTypeId() As Integer
        Get
            Return m_OrderTypeId
        End Get
        Set(value As Integer)
            m_OrderTypeId = Value
        End Set
    End Property
    Private m_OrderTypeId As Integer

    Public Overridable Property Orders() As ICollection(Of Order)
        Get
            Return Me.m_orders
        End Get
        Set(value As ICollection(Of Order))
            Me.m_orders = value
        End Set
    End Property

    Public Overridable Property Shippers() As ICollection(Of Shipper)
        Get
            Return Me.m_shippers
        End Get
        Set(value As ICollection(Of Shipper))
            Me.m_shippers = value
        End Set
    End Property

    Public Property Description() As String
        Get
            Return m_Description
        End Get
        Set(value As String)
            m_Description = Value
        End Set
    End Property
    Private m_Description As String
End Class
